codegen/method.go

```diff
@@ -895,18 +895,16 @@ func (ms *MethodSpec) setTypeConverters(
 		*compile.DoubleSpec,
 		*compile.StringSpec:
 
-		respConverter.append("out", " := in\t\n")
+		isPrimitive = true
 	default:
-		// default as struct
 		respFields = respType.(*compile.StructSpec).Fields
 		downstreamRespFields = downstreamRespType.(*compile.StructSpec).Fields
-		respConverter.append("out", " := ", "&", ms.ShortResponseType, "{}\t\n")
-		err = respConverter.GenStructConverter(downstreamRespFields, respFields, respTransforms)
-		if err != nil {
-			return err
-		}
+		isPrimitive = false
+	}
+	err = respConverter.GenStructConverter(downstreamRespFields, respFields, respTransforms, isPrimitive)
```

Comment:
`isPrimitive` boolean seems ugly. Can we just pass in `respType` instead of `respFields` ?